src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -32,81 +32,481 @@
+
+                if (isDefaultFieldMappingSet) {
+                    BDI_HelpTextCMT.DataImportFieldMapping clonedFieldMapping =
+                        new BDI_HelpTextCMT.DataImportFieldMapping(dataImportFieldMapping,
+                            MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME);
+                        queuedMetadataTypesForDeploy.add(clonedFieldMapping.getMetadataRecord());
+                } else {
+                    updatedConflictingMappings.add(dataImportFieldMapping);
+                }
+            }
+
+            String fieldMappingSetToCompareAgainst;
+            if (BDI_MigrationMappingUtility.customFieldMappingSetExists == true) {
+                fieldMappingSetToCompareAgainst = MIGRATED_CUSTOM_FIELD_MAPPING_SET_RECORD_NAME;
+            } else {
+                fieldMappingSetToCompareAgainst = DEFAULT_FIELD_MAPPING_SET_RECORD_NAME;
+            }
+
+            List<Metadata.CustomMetadata> updatedCMTs = updateCustomMetadataRecords(fieldMappingSetToCompareAgainst,
+                updatedConflictingMappings);
+
+            queuedMetadataTypesForDeploy.addAll(updatedCMTs);
+        }
+    }
+
+    /**
+    * @description Method updates existing custom metadata records in memory.
+    * @param fieldMappingSetName Name of parent Field Mapping Set
+    * @param fieldMappings List of field mappings with new field values we want to keep
+    * @return List of custom metadata type records
+    */
+    public List<Metadata.CustomMetadata> updateCustomMetadataRecords(
+        String fieldMappingSetName,
+        List<BDI_HelpTextCMT.DataImportFieldMapping> fieldMappings) {
+
+        Map<String, Data_Import_Field_Mapping__mdt> fieldMappingsByFieldMappingSetDeveloperName =
+            helper.getFieldMappingsByFieldMappingSetDeveloperName(fieldMappingSetName);
+        List<Metadata.CustomMetadata> updatedCMTs = new List<Metadata.CustomMetadata>();
+
+        for (BDI_HelpTextCMT.DataImportFieldMapping fieldMapping : fieldMappings) {
+            Data_Import_Field_Mapping__mdt existingDefaultFieldMapping =
+                fieldMappingsByFieldMappingSetDeveloperName.get(fieldMapping.sourceFieldAPIName);
+
+            Metadata.CustomMetadata updatedCMT = fieldMapping.getMetadataRecord();
+            List<String> nameParts = updatedCMT.fullName.split('\\.', 2);
+            updatedCMT.fullName = nameParts[0] + '.' + existingDefaultFieldMapping.DeveloperName;
+            updatedCMTs.add(updatedCMT);
+        }
+
+        return updatedCMTs;
+    }
+
+    /**
+    * @description Method for handling special cased mappings (GAU and Opportunity Contact Role).
+    * Since GAUs and Opportunity Contact Roles aren't mapped in helptext, we need to inspect and collect
+    * our existing packaged data import field mappings in order to clone them for the migrated custom
+    * field mapping set.
+    */
+    public void handleGAUAndOpportunityContactRoleMappings() {
```

Comment:
I wanted to dive in to this a bit more deeply before the end of the day, but could we approach this more like "for any object mappings that don't exist in the help text mapping, bring over all the corresponding field mappings"? My concern here is that a) it's a bit hard-coded to OCR and GAU alloc and b) we might add new default mappings without help text mapping equivalent and it would be nice to bring those in during migration without having to adjust the code. 

Second thought, do we just want to copy in any default CMT field mappings that don't clash with their help text mappings (Including GAU and OCR) to create the new customized set? I don't want to surprise customers with changed config, but am I right in thinking that adding new field mappings should be reasonably safe?

Reply:
100% makes total sense. I think it absolutely makes sense to bring over any and all default field mappings that don't clash with their help text mappings. Pretty much what I wanted to do, it just so happens in this case everything is mapped via help text _except_ GAUs and OCRs. I have a good idea of where I can update this 👍